proxy/coco/tests/working_copy.rs

```diff
@@ -0,0 +1,57 @@
+use coco::RunConfig;
+
+use pretty_assertions::assert_eq;
+
+#[macro_use]
+mod common;
+use common::{build_peer, init_logging, shia_le_pathbuf};
+
+#[tokio::test]
+async fn upstream_for_default() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Sorry for my ignorance, but I don't quite understand what this is testing. The name would suggest that "upstream" would be tested as a project's default branch but that doesn't seem to be it.

Reply:
> Sorry for my ignorance, but I don't quite understand what this is testing

Perfectly valid question.

There's a concept of an upstream or tracking branches in git -- see here https://git-scm.com/book/en/v2/Git-Branching-Remote-Branches.

We're confirming the behaviour of setting the upstream branch since we had to work around `git2` in this scenario and write to the git config directly.

Reply:
Ah, I see! Cool, the test now makes sense to me :) thank you!